patches/node/fix_crypto_tests_to_run_with_bssl.patch

```diff
@@ -1010,3 +1010,36 @@ index 1094845c73e14313860ad476fb7baba2a11b5af4..51972b4b34b191ac59145889dbf2da5c
  };
  
  function generateWrappingKeys() {
+diff --git a/test/parallel/test-x509-escaping.js b/test/parallel/test-x509-escaping.js
+index 99418e4c0bf21c26d5ba0ad9d617419abc625593..fc129b26ea13895353d6ede26bb2d91695c94ba4 100644
+--- a/test/parallel/test-x509-escaping.js
++++ b/test/parallel/test-x509-escaping.js
+@@ -425,11 +425,11 @@ const { hasOpenSSL3 } = common;
+   assert.strictEqual(certX509.subjectAltName, 'DNS:evil.example.com');
+ 
+   // The newer X509Certificate API allows customizing this behavior:
+-  assert.strictEqual(certX509.checkHost(servername), servername);
++  assert.strictEqual(certX509.checkHost(servername), undefined);
```

Comment:
This is interesting. What's this about @jkleinsc

Reply:
@ckerr I added that due to differences between boringssl and openssl.

Reply:
@jkleinsc is this ... the right change though? It seems to me like this might create Problems for people using this API. Can we fix it by changing the underlying BSSL wrapper to return the right value?

Reply:
@nornagon I think that would mean reverting this change from 2019 where boringssl deprecated the flag ` X509_CHECK_FLAG_ALWAYS_CHECK_SUBJECT` that node is using 
https://boringssl.googlesource.com/boringssl/+/923feba60897676019b00d017ae4eba796a1f6ff

Reply:
FWIW... I just tested reverting these changes and it does indeed make the (unmodified) node test pass.  So I think the only question is whether or not its a security risk to revert the above CLs?

Reply:
@nornagon what if we add a new flag that turns on the old behavior?

Reply:
I don't understand, do you mean a build flag? A runtime flag? Where would the flag be passed? Generally that doesn't sound like a good idea, we should have one consistent behavior.

Reply:
@nornagon a runtime flag [do_x509_check](https://source.chromium.org/chromium/chromium/src/+/main:third_party/boringssl/src/crypto/x509v3/v3_utl.c;l=1009) accepts flags as a parameter.  When calling from node we could add the additional flag to ensure expected behavior from there without affecting behavior elsewhere.  I'm open to other suggestions though.

FWIW... here's what the patch looks like to put back the removed functionality (I didn't add the new flag though):
```
diff --git a/crypto/x509v3/v3_utl.c b/crypto/x509v3/v3_utl.c
index 960c407a1..8d8e87fed 100644
--- a/crypto/x509v3/v3_utl.c
+++ b/crypto/x509v3/v3_utl.c
@@ -1015,6 +1015,7 @@ static int do_x509_check(X509 *x, const char *chk, size_t chklen,
     int j;
     int cnid = NID_undef;
     int alt_type;
+    int san_present = 0;
     int rv = 0;
     equal_fn equal;

@@ -1042,6 +1043,7 @@ static int do_x509_check(X509 *x, const char *chk, size_t chklen,
             gen = sk_GENERAL_NAME_value(gens, i);
             if (gen->type != check_type)
                 continue;
+            san_present = 1;
             if (check_type == GEN_EMAIL)
                 cstr = gen->d.rfc822Name;
             else if (check_type == GEN_DNS)
@@ -1054,7 +1056,10 @@ static int do_x509_check(X509 *x, const char *chk, size_t chklen,
                 break;
         }
         GENERAL_NAMES_free(gens);
-        return rv;
+        if (rv != 0)
+            return rv;
+        if (san_present && !(flags & X509_CHECK_FLAG_ALWAYS_CHECK_SUBJECT))
+            return 0;
     }

     /* We're done if CN-ID is not pertinent */
diff --git a/include/openssl/x509v3.h b/include/openssl/x509v3.h
index c67dde6a3..b154b7342 100644
--- a/include/openssl/x509v3.h
+++ b/include/openssl/x509v3.h
@@ -886,8 +886,8 @@ OPENSSL_EXPORT void X509_email_free(STACK_OF(OPENSSL_STRING) *sk);
 OPENSSL_EXPORT STACK_OF(OPENSSL_STRING) *X509_get1_ocsp(X509 *x);
 // Flags for X509_check_* functions

-// Deprecated: this flag does nothing
-#define X509_CHECK_FLAG_ALWAYS_CHECK_SUBJECT 0
+// Always check subject name for host match even if subject alt names present
+#define X509_CHECK_FLAG_ALWAYS_CHECK_SUBJECT   0x1
 // Disable wildcard matching for dnsName fields and common name.
 #define X509_CHECK_FLAG_NO_WILDCARDS 0x2
 // X509_CHECK_FLAG_NO_PARTIAL_WILDCARDS does nothing, but is necessary in
```

Reply:
Thanks for the insight @davidben! I'm happy to go with disabling the test given the above.